### PR TITLE
Add Keycloak Proxy chart

### PR DESCRIPTION
**Pull request by YunSangJun**
**What this PR does / why we need it**:
[Why do we need Keycloak Proxy chart]
There is already a Keycloak Proxy docker image on the Docker Hub. But it takes a lot of effort to deploy it on Kubernetes cluster. This Helm chart helps you easily deploy the Keycloak Proxy application to the Kubernetes cluster.

[Why do we need Keycloak Proxy?]
It is often necessary to work with the OpenID Connect Provider to authenticate the application. However, in some cases, the application does not provide a plug-in for interoperating with the OpenID Connect Provider, and the source code can not be customized (for example, Kibana). In this case, if you use Keycloak Proxy, you can work with OpenID Connect Provider with sidecar pattern.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by YunSangJun** [comment]
/assign @sameersbn

I've got the following error message from CircleCI. 
I don't know what these means.
```
Linting the Chart.yaml and values.yaml files at incubator/keycloak-proxy
incubator/keycloak-proxy/values.yaml
  1:1       error    too many blank lines (1 > 0)  (empty-lines)
```

File: incubator/keycloak-proxy/values.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+
```
Comment (timstoop) [inline incubator/keycloak-proxy/values.yaml:4]: If you remove this empty line, CI will probably succeed.

**Review by timstoop — CHANGES_REQUESTED** [review]

File: incubator/keycloak-proxy/values.yaml (hunk 1 above)
Comment (YunSangJun) [reply incubator/keycloak-proxy/values.yaml:4]: @unguiculus Thank you. The test was passed.

File: incubator/keycloak-proxy/Chart.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+name: keycloak-proxy
+version: 0.0.1
+appVersion: 3.4.0.Final
```
Comment (unguiculus) [inline incubator/keycloak-proxy/Chart.yaml]: Why not use 3.4.2.Final?

File: incubator/keycloak-proxy/README.md [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+Alternatively, a YAML file that specifies the values for the parameters can be provided while installing the chart. For example,
+
+```bash
+$ helm install --name keycloak-proxy -f values.yaml incubator/keycloak-proxy
+```
+
+## Proxy Configuration
+
+The following configurations which are located in a configmap are required to request authentication and authorization.
+Please refer to [Keycloak Proxy](https://www.keycloak.org/docs/3.3/server_installation/topics/proxy.html) and [Adapter Config](https://www.keycloak.org/docs/3.3/securing_apps/topics/oidc/java/java-adapter-config.html) for more information.
```
Comment (unguiculus) [inline incubator/keycloak-proxy/README.md]: Doc links refer to older version.

**Review by unguiculus — CHANGES_REQUESTED** [review]
Please compare with what you get with `helm create` using Helm 2.8+ an update accordingly. Also, make sure you follow our review guidelines and best practices.

File: incubator/keycloak-proxy/Chart.yaml (hunk 2 above)
Comment (YunSangJun) [reply incubator/keycloak-proxy/Chart.yaml]: @unguiculus That's my mistake. I fixed it and tested.

**Comment by YunSangJun** [comment]
@unguiculus I updated this chart to correspond with Helm 2.8+ version and guidelines and best practices. Please review and merge.

**Comment by YunSangJun** [comment]
@timstoop @sameersbn @unguiculus Can anyone review this?.

File: incubator/keycloak-proxy/README.md [hunk 4]
```diff
@@ -11,9 +11,9 @@ $ helm install incubator/keycloak-proxy
 
 ## Introduction
 
-This chart bootstraps a [Keycloak Proxy](https://www.keycloak.org/docs/3.3/server_installation/topics/proxy.html) Deployment on a [Kubernetes](https://kubernetes.io) cluster
+This chart bootstraps a [Keycloak Proxy](https://www.keycloak.org/docs/latest/server_installation/index.html#_proxy) Deployment on a [Kubernetes](https://kubernetes.io) cluster
```
Comment (unguiculus) [inline incubator/keycloak-proxy/README.md]: Doc links should refer to the 3.4 docs.

File: incubator/keycloak-proxy/README.md [hunk 5]
```diff
@@ -72,7 +77,7 @@ $ helm install --name keycloak-proxy -f values.yaml incubator/keycloak-proxy
 ## Proxy Configuration
 
 The following configurations which are located in a configmap are required to request authentication and authorization.
-Please refer to [Keycloak Proxy](https://www.keycloak.org/docs/3.3/server_installation/topics/proxy.html) and [Adapter Config](https://www.keycloak.org/docs/3.3/securing_apps/topics/oidc/java/java-adapter-config.html) for more information.
+Please refer to [Keycloak Proxy](https://www.keycloak.org/docs/latest/server_installation/index.html#_proxy) and [Adapter Config](https://www.keycloak.org/docs/latest/securing_apps/index.html#_java_adapter_config) for more information.
```
Comment (unguiculus) [inline incubator/keycloak-proxy/README.md]: Doc links should refer to the 3.4 docs.

File: incubator/keycloak-proxy/templates/deployment.yaml [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+            - name: http
+              containerPort: 8080
+              protocol: TCP
+          resources:
+{{ toYaml .Values.resources | indent 12 }}
+          volumeMounts:
+            - mountPath: /opt/jboss/conf
+              name: proxy-config
+      volumes:
+      - name: proxy-config
```
Comment (unguiculus) [inline incubator/keycloak-proxy/templates/deployment.yaml]: Indent list here one more level for consistency.

**Review by unguiculus — CHANGES_REQUESTED** [review]

File: incubator/keycloak-proxy/README.md (hunk 4 above)
Comment (YunSangJun) [reply incubator/keycloak-proxy/README.md]: I fixed it.

File: incubator/keycloak-proxy/README.md (hunk 5 above)
Comment (YunSangJun) [reply incubator/keycloak-proxy/README.md]: I fixed it.

File: incubator/keycloak-proxy/templates/deployment.yaml (hunk 6 above)
Comment (YunSangJun) [reply incubator/keycloak-proxy/templates/deployment.yaml]: I fixed it.

**Comment by YunSangJun** [comment]
@unguiculus Thank you for review. I fixed it.

**Comment by YunSangJun** [comment]
@timstoop @sameersbn @unguiculus I fixed the requested changes. Please review and merge.

**Comment by YunSangJun** [comment]
@timstoop @sameersbn @unguiculus I've waited a long time and would like to ask you again. Please review it and merge it if there is no problem.

**Review by unguiculus — APPROVED** [review]

**Pull request merged**